processing/src/main/java/org/apache/carbondata/processing/sortandgroupby/sortdata/SortDataRows.java

```diff
@@ -375,6 +376,9 @@ private void writeDataWithOutKettle(Object[][] recordHolderList, int entryCountL
             stream.write((byte) 0);
           }
         }
+        if (i % 10000 == 9999) {
```

Comment:
it is unnecessary calculation we are adding in it. Already the buffer size is configurable so once buffer size reaches buffered out stream can flush it down.